ui/src/chatbot.tsx

```diff
@@ -106,6 +114,7 @@ export const XChatbot = (props: Chatbot) => {
     [msgs, setMsgs] = React.useState<Message[]>(props.data ? processData(props.data) : []),
     [userInput, setUserInput] = React.useState(''),
     [isInfiniteLoading, setIsInfiniteLoading] = React.useState(false),
+    [rated, setRated] = React.useState(() => new Map()),
```

Comment:
Why is this a function? The map could also use some types. Do we need an external map though? Couldn't the  feedback state be part of the message instead?